pyam/core.py

```diff
@@ -707,7 +707,14 @@ def timeseries(self, iamc_index=False):
         if self.empty:
             raise ValueError("This IamDataFrame is empty!")
 
-        df = self._data.unstack(level=self.time_col).rename_axis(None, axis=1)
+        s = self._data
+        if iamc_index:
+            if self.time_col == "time":
+                raise ValueError(
+                    "Cannot use IAMC-index with continuous-time data format!"
+                )
+            s = s.droplevel(self.extra_cols)
+        df = s.unstack(level=self.time_col).rename_axis(None, axis=1).sort_index(axis=1)
 
         if df.index.has_duplicates:
```

Comment:
Based on your earlier comment that checking for duplicates is expensive, and if we assume that the `self_data` has a unique index to begin with, this check only needs to happen if `iamc_index=True`...

Reply:
It is expensive the first time you do it, but it is also cached! So you should mainly avoid doing it in a loop! Since we typically verified it after construction, it's basically free here:

<img width="482" alt="image" src="https://user-images.githubusercontent.com/2552981/110836278-2873fb80-82a0-11eb-8ca5-0c7fa34416e5.png">

Reply:
Also, easy to test: Result -> Nothing to worry about.

<img width="497" alt="image" src="https://user-images.githubusercontent.com/2552981/110837237-44c46800-82a1-11eb-97dc-c078ad84ad02.png">

Reply:
got it, thanks!